Add public API endpoint for public collections

tw4l opened the pull request:
Fixes #1051 

If org with provided slug doesn't exist or no public collections exist for that org, return same 404 response with a detail of "public_profile_not_found" to prevent people from using public endpoint to determine whether an org exists.

Endpoint is `GET /api/public-collections/<org-slug>` (no auth needed) to avoid collisions with existing org and collection endpoints.

@SuaYoo happy to make any changes needed for the frontend, just lmk!

tw4l committed:
Add public API endpoint for public collections

If org with provided slug doesn't exist or no public collections
exist for that org, return same 404 response with a detail of
"public_collections_not_found" to prevent people from using public
endpoint to determine whether an org exists.

Endpoint is /api/public-collections/<org-slug> to avoid collisions
with existing org and collection endpoints.

SuaYoo commented on backend/btrixcloud/colls.py:
```diff
@@ -395,6 +397,23 @@ async def add_successful_crawl_to_collections(self, crawl_id: str, cid: UUID):
+            raise HTTPException(status_code=404, detail="public_collections_not_found")
+
+        collections, _ = await self.list_collections(
+            org.id, access=CollAccessType.PUBLIC
```
I believe we should be checking if `listPublicCollections` is enabled here, rather than if there's any public collections.

[Per Discord convo](https://discord.com/channels/895426029194207262/920450066592968734/1310659058096668702), we might want to rename this to a more general `enablePublicProfile` or similar name since the org profile page may be enabled even without any public collections.

SuaYoo replied:
What about responding with a `200` and `null` data rather than 404? I know we're not exactly following the [JSON API spec](https://jsonapi.org/format/#fetching-resources-responses-200) but as an API consumer the following responses seem both intuitive and generic enough for security reasons, and wouldn't need custom 404 handling that checks error details:

Response for nonexistent org, or org with `listPublicCollections: false`:
```
GET /public-collections/<org-slug>

Response: 200
{
  "org": null
}
```

Response for org with `listPublicCollections: true` (empty array if no public collections):
```
GET /public-collections/<org-slug>

Response: 200
{
  "org": {
    "name": "<org name>",
    "collections": []
  }
}
```

tw4l replied:
> I believe we should be checking if `listPublicCollections` is enabled here, rather than if there's any public collections.

Ah yeah, I missed adding a check for `listPublicCollections` (or its equivalent if we rename) before adding the collections filtered on `CollAccessType.PUBLIC`, will add that.

SuaYoo commented on backend/btrixcloud/colls.py:
```diff
@@ -582,4 +601,12 @@ async def download_collection(
         return await colls.download_collection(coll_id, org)
 
+    @app.get(
+        "/public-collections/{org_slug}",
```
Is this endpoint rate limited?

tw4l replied:
None of our endpoints currently are, we might want to handle that in nginx?

tw4l committed:
Rename listPublicCollections to enablePublicProfile

tw4l committed:
Only list public collections if public profile enabled

tw4l committed:
Remove duplicate tests

tw4l committed:
Use model for public org details in public list endpoint

This will be more easily extensible if we want to add additional
fields (like an organization description) that would be visible in
a public profile.

tw4l commented:
Converting to draft while I make changes

tw4l commented:
Made all of the changes except the 404 -> 200 change. Also restructured the response a bit to look like:

```
{
  "org": {
    "name": "Org name",
    // Can easily add other fields here moving forward
  },
  "collections": [
    {"id": "some uuid", "name": "A public collection", "access": "public",...},
    ...
  ]
}
```

tw4l commented:
Also tagged @ikreymer to see if he has thoughts/comments/suggested changes (overall and/or on the 404 question)

tw4l committed:
Add support for public org description

tw4l committed:
Fix key for description in public colls endpoint

tw4l committed:
Format with Black

tw4l commented:
Added support for a public org description (editing via same /public-profile endpoint that's used to enable/disable public profile), as requested in https://github.com/webrecorder/browsertrix/issues/1051#issuecomment-2499156744

tw4l committed:
Fix typo in test key

tw4l committed:
Return public profile info even if no public collections yet

tw4l committed:
Add url to org public profile and public endpoint

tw4l committed:
Add test case for existing org with public profile disabled

tw4l committed:
Add test for public profile with no collections

tw4l commented:
Some additional changes made:
- Public profiles can now include a URL field
- /public-collections endpoint still returns data if org has public profile enabled even if there are no public collections yet
- 404 message for /public-collections endpoint modified to `public_profile_not_found`

tw4l committed:
Fix test that still expected old 404 detail

tw4l committed:
Add url to PublicOrgDetails model

tw4l committed:
Include url in public org details in response

tw4l committed:
Don't pass None to str validator

tw4l committed:
Fix side effect of using non default org in earlier test

SuaYoo reviewed (approved):
Looks good! Tested with https://github.com/webrecorder/browsertrix/pull/2172

tw4l commented:
Merging into feature branch, can do another review before merging to main
